File: SYP.Web/Modules/ServerTypes/Warehouse/StockExitsColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { StockExitsRow } from "./StockExitsRow";
import { StockExitStatus } from "./StockExitStatus";

export interface StockExitsColumns {
    ExitNo: Column<StockExitsRow>;
    WarehouseName: Column<StockExitsRow>;
    ExitDate: Column<StockExitsRow>;
    Status: Column<StockExitsRow>;
    Description: Column<StockExitsRow>;
    InsertDate: Column<StockExitsRow>;
}

export class StockExitsColumns extends ColumnsBase<StockExitsRow> {
    static readonly columnsKey = 'Warehouse.StockExits';
    static readonly Fields = fieldsProxy<StockExitsColumns>();
}

[StockExitStatus]; // referenced types

File: SYP.Web/Modules/Warehouse/StockExits/StockExitsColumns.cs

namespace SYP.Warehouse.Columns;

[ColumnsScript("Warehouse.StockExits")]
[BasedOnRow(typeof(StockExitsRow), CheckNames = true)]
public class StockExitsColumns
{
    [EditLink, Width(120)]
    public string ExitNo { get; set; }

    [Width(150)]
    public string WarehouseName { get; set; }

    [Width(150), DisplayFormat("dd.MM.yyyy HH:mm")]
    public DateTime ExitDate { get; set; }

    [Width(100)]
    public StockExitStatus Status { get; set; }

    [Width(250)]
    public string Description { get; set; }

    [Width(150), DisplayFormat("dd.MM.yyyy HH:mm")]
    public DateTime InsertDate { get; set; }
}
